packages/detector_test.go

```diff
@@ -71,6 +72,15 @@ func TestHelperProcess(t *testing.T) {
 				fmt.Fprintln(os.Stdout, ">>> APK CODE : ASKING FOR SOMETHING ELSE ... ITS NOT INSTALLED <<<<")
 				os.Exit(1)
 			}
+		} else if os.Getenv("GO_WANT_HELPER_PROCESS") == "dnfinstalled" {
+			fmt.Println(">>> DNF CODE <<<<")
+			if expectedProgram == "dnf" {
+				fmt.Fprintln(os.Stdout, ">>> DNF CODE : ASKING FOR DNF ... ITS INSTALLED <<<<")
+				os.Exit(0)
+			} else {
+				fmt.Fprintln(os.Stdout, ">>> DNF CODE : ASKING FOR SOMETHING ELSE ... ITS NOT INSTALLED <<<<")
+				os.Exit(1)
```

Comment:
I would shy away from using `os.Exit` in tests, and just use `t.Error`

Reply:
yah this test is special. Since auto detect works by exec'ing `which` the only way i can test it is to actually have an exec run. And in this case if the actual code asks for something other than `dnf` i want the exec command to fail with a non zero exit code. 

https://jamiethompson.me/posts/Unit-Testing-Exec-Command-In-Golang/